js/src/components/ads-account-select-control/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import AppSelectControl from '.~/components/app-select-control';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+
+/**
+ * @param {Object} props The component props
+ * @param {string} [props.value] The selected value. IF no value is defined, then the first option is selected and onChange function is triggered.
+ * @param {Function} [props.onChange] Callback when the select value changes.
+ * @return {JSX.Element} An enhanced AppSelectControl component.
+ */
+const AdsAccountSelectControl = ( {
+	value,
+	onChange = () => {},
+	...props
+} ) => {
+	const { existingAccounts: accounts = [] } = useExistingGoogleAdsAccounts();
+
+	const options = accounts.map( ( acc ) => ( {
+		value: acc.id,
+		label: sprintf(
+			// translators: 1: account name, 2: account ID.
+			__( '%1$s (%2$s)', 'google-listings-and-ads' ),
+			acc.name,
+			acc.id
+		),
+	} ) );
+
+	useEffect( () => {
+		// Triggers the onChange event in order to pre-select the initial value
+		if ( value === undefined ) {
```

Comment:
```suggestion
		if ( value === undefined && options.length > 0 ) {
```

We may need to handle this as well.

Reply:
@ankitrox on L39, there's already an optional chaining operator to handle it. Also it's consistent to what we already have in `js/src/components/merchant-center-select-control/index.js`